src/nupic/engine/Network.cpp

```diff
@@ -363,7 +363,9 @@ Network::link(const std::string& srcRegionName, const std::string& destRegionNam
   }
 
   // Create the link itself
-  destInput->addLink(linkType, linkParams, srcOutput, propagationDelay);
+  auto link = new Link(linkType, linkParams, srcOutput, destInput,
+                       propagationDelay);
+  destInput->addLink(link, srcOutput);
```

Comment:
This change and the one below are minor code cleanup that take advantage of the new separation of logic for instantiating a Link versus adding the Link to the Input.